.github/workflows/main.yml

```diff
@@ -0,0 +1,64 @@
+name: Main workflow
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest
+          #- windows-latest
+        ocaml-version:
+          - 4.09.1
+          - 4.08.1
+          - 4.07.1
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Use OCaml ${{ matrix.ocaml-version }}
+        uses: avsm/setup-ocaml@v1
+        with:
+          ocaml-version: ${{ matrix.ocaml-version }}
+
+      - name: Pin BAP
+        run: opam pin add bap . --no-action
+
+      - name: Configure Homebrew LLVM
+        if: matrix.os == 'macos-latest'
+        run: echo '::set-env name=LLVM_CONFIG::/usr/local/opt/llvm/bin/llvm-config'
+
+      - name: Install system dependencies
+        run: opam depext bap --yes
+
+      - name: Install opam dependencies
+        run: opam install . --deps-only
+
+      - name: Install BAP
+        run: opam install bap -v --yes --with-test
+        env:
+          XDG_CACHE_HOME: .bap-cache
```

Comment:
looks like that we're blocked by #1109 and the fact that this variable is actually used on MacOS by other applications. 
As a temporary workaround, let's remove these two lines and change `make test` in opam/opam to `XDG_CACHE_HOME=.bap-cache make test` to limit the effect of this variable.